### PR TITLE
PLT-5932 Split out action_types from constants

### PR DESCRIPTION
**Pull request by jwilander**
#### Summary
* Split out action_types from constants
* Remove `s` for type names (ex. `UsersTypes`  to `UserTypes`)
* Separate some constants out into different files

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-5932

#### Checklist
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

File: src/constants/general.js [hunk 1]
```diff
@@ -1,38 +1,36 @@
...
+export default {
+    CONFIG_CHANGED: 'config_changed',
 
-const GeneralTypes = keyMirror({
-    RECEIVED_APP_STATE: null,
-    RECEIVED_APP_CREDENTIALS: null,
-    REMOVED_APP_CREDENTIALS: null,
-    RECEIVED_APP_DEVICE_TOKEN: null,
+    PROFILE_CHUNK_SIZE: 100,
+    CHANNELS_CHUNK_SIZE: 50,
```
Comment (enahum) [inline src/constants/general.js]: If we have a file for `Posts` constants that include the `POSTS_CHUNK_SiZE` why not add a file for users (profiles and channels for this two?

File: src/constants/general.js [hunk 2]
```diff
@@ -1,38 +1,36 @@
...
-    PING_REQUEST: null,
-    PING_SUCCESS: null,
-    PING_FAILURE: null,
-    PING_RESET: null,
+    MENTION: 'mention',
 
-    RECEIVED_SERVER_VERSION: null,
+    OFFLINE: 'offline',
+    AWAY: 'away',
+    ONLINE: 'online',
```
Comment (enahum) [inline src/constants/general.js]: we can move this three to the user constant file

File: src/constants/general.js [hunk 3]
```diff
@@ -1,38 +1,36 @@
...
 
-    CLIENT_CONFIG_REQUEST: null,
-    CLIENT_CONFIG_SUCCESS: null,
-    CLIENT_CONFIG_FAILURE: null,
-    CLIENT_CONFIG_RECEIVED: null,
+    TEAM_USER_ROLE: 'team_user',
+    TEAM_ADMIN_ROLE: 'team_admin',
```
Comment (enahum) [inline src/constants/general.js]: we can create another one for team constants

File: src/constants/general.js [hunk 4]
```diff
@@ -1,38 +1,36 @@
...
 
-    CLIENT_LICENSE_REQUEST: null,
-    CLIENT_LICENSE_SUCCESS: null,
-    CLIENT_LICENSE_FAILURE: null,
-    CLIENT_LICENSE_RECEIVED: null,
+    CHANNEL_USER_ROLE: 'channel_user',
+    CHANNEL_ADMIN_ROLE: 'channel_admin',
```
Comment (enahum) [inline src/constants/general.js]: move these to the channel constants

File: src/constants/general.js [hunk 5]
```diff
@@ -1,38 +1,36 @@
...
+    SYSTEM_ADMIN_ROLE: 'system_admin',
 
-    WEBSOCKET_REQUEST: null,
-    WEBSOCKET_SUCCESS: null,
-    WEBSOCKET_FAILURE: null
-});
+    DEFAULT_CHANNEL: 'town-square',
+    DM_CHANNEL: 'D',
+    OPEN_CHANNEL: 'O',
+    PRIVATE_CHANNEL: 'P',
```
Comment (enahum) [inline src/constants/general.js]: same here

File: src/constants/general.js [hunk 6]
```diff
@@ -1,38 +1,36 @@
...
 
-export default GeneralTypes;
+    START_OF_NEW_MESSAGES: 'start-of-new-messages',
```
Comment (enahum) [inline src/constants/general.js]: this one to channel?

File: src/constants/general.js (hunk 1 above)
Comment (jwilander) [inline src/constants/general.js]: Yes I could but I've already spent enough time on refactoring constants and I think that mana would be better spent on the driver and the store itself

**Comment by enahum** [comment]
Alright we can move them later on in a smaller ticket

**Comment by jwilander** [comment]
Cool, I'll make another ticket for it

**Pull request merged**